processing/src/main/java/io/druid/query/extraction/DimExtractionFn.java

```diff
@@ -1,73 +1,35 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0
  *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
  */
 
 package io.druid.query.extraction;
 
-import com.fasterxml.jackson.annotation.JsonSubTypes;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
-
-/**
- */
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
-@JsonSubTypes(value = {
-    @JsonSubTypes.Type(name = "time", value = TimeDimExtractionFn.class),
-    @JsonSubTypes.Type(name = "regex", value = RegexDimExtractionFn.class),
-    @JsonSubTypes.Type(name = "partial", value = MatchingDimExtractionFn.class),
-    @JsonSubTypes.Type(name = "searchQuery", value = SearchQuerySpecDimExtractionFn.class),
-    @JsonSubTypes.Type(name = "javascript", value = JavascriptDimExtractionFn.class)
-})
-/**
- * A DimExtractionFn is a function that can be used to modify the values of a dimension column.
- *
- * A simple example of the type of operation this enables is the RegexDimExtractionFn which applies a
- * regular expression with a capture group.  When the regular expression matches the value of a dimension,
- * the value captured by the group is used for grouping operations instead of the dimension value.
- */
-public interface DimExtractionFn
+public abstract class DimExtractionFn implements ExtractionFn
 {
-  /**
-   * Returns a byte[] unique to all concrete implementations of DimExtractionFn.  This byte[] is used to
-   * generate a cache key for the specific query.
-   *
-   * @return a byte[] unit to all concrete implements of DimExtractionFn
-   */
-  public byte[] getCacheKey();
-
-  /**
-   * The "extraction" function.  This should map a dimension value into some other value.
-   *
-   * In order to maintain the "null and empty string are equivalent" semantics that Druid provides, the
-   * empty string is considered invalid output for this method and should instead return null.  This is
-   * a contract on the method rather than enforced at a lower level in order to eliminate a global check
-   * for extraction functions that do not already need one.
-   *
-   *
-   * @param dimValue the original value of the dimension
-   * @return a value that should be used instead of the original
-   */
-  public String apply(String dimValue);
+  @Override
+  public String apply(Object value)
+  {
+    return apply(value.toString());
```

Comment:
Does the `toString()` increase query time for String based extraction functions?